src/runtime/virtcontainers/remote.go

```diff
@@ -0,0 +1,287 @@
+
+	socketPath := rh.agentSocketPath
+	if len(socketPath) == 0 {
+		return nil, errors.New("failed to generate remote sock: TunnelSocketPath is not set")
+	}
+
+	remoteSock := types.RemoteSock{
+		SandboxID:        id,
+		TunnelSocketPath: socketPath,
+	}
+
+	return remoteSock, nil
+}
+
+func notImplemented(name string) error {
+
+	err := errors.Errorf("%s: not implemented", name)
+
+	logrus.Errorf(err.Error())
+
+	if tracer, ok := err.(interface{ StackTrace() errors.StackTrace }); ok {
+		for _, f := range tracer.StackTrace() {
+			logrus.Errorf("%+s:%d\n", f, f)
+		}
+	}
+
+	return err
+}
+
+func (rh *remoteHypervisor) PauseVM(ctx context.Context) error {
+	panic(notImplemented("PauseVM"))
+}
+
+func (rh *remoteHypervisor) SaveVM() error {
+	panic(notImplemented("SaveVM"))
+}
+
+func (rh *remoteHypervisor) ResumeVM(ctx context.Context) error {
+	panic(notImplemented("ResumeVM"))
+}
+
+func (rh *remoteHypervisor) AddDevice(ctx context.Context, devInfo interface{}, devType DeviceType) error {
+	// TODO
+	logrus.Printf("addDevice: deviceType=%v devInfo=%#v", devType, devInfo)
+	return nil
+}
+
+func (rh *remoteHypervisor) HotplugAddDevice(ctx context.Context, devInfo interface{}, devType DeviceType) (interface{}, error) {
+	logrus.Printf("HotplugAddDevice: devInfo=%#v", devInfo)
+	return "HotplugAddDevice is not implemented", nil
+}
+
+func (rh *remoteHypervisor) HotplugRemoveDevice(ctx context.Context, devInfo interface{}, devType DeviceType) (interface{}, error) {
+	logrus.Printf("HotplugRemoveDevice: devInfo=%#v", devInfo)
+	return "HotplugRemoveDevice is not implemented", nil
+}
+
+func (rh *remoteHypervisor) ResizeMemory(ctx context.Context, memMB uint32, memoryBlockSizeMB uint32, probe bool) (uint32, MemoryDevice, error) {
+	// TODO
+	return memMB, MemoryDevice{}, nil
+}
+
+func (rh *remoteHypervisor) ResizeVCPUs(ctx context.Context, vcpus uint32) (uint32, uint32, error) {
+	// TODO
+	return vcpus, vcpus, nil
+}
+
+func (rh *remoteHypervisor) GetVMConsole(ctx context.Context, sandboxID string) (string, string, error) {
+	panic(notImplemented("GetVMConsole"))
+}
+
+func (rh *remoteHypervisor) Disconnect(ctx context.Context) {
+	// TODO
+	panic(notImplemented("Disconnect"))
+}
+
+func (rh *remoteHypervisor) Capabilities(ctx context.Context) types.Capabilities {
+	return types.Capabilities{}
+}
+
+func (rh *remoteHypervisor) HypervisorConfig() HypervisorConfig {
+	return rh.config
+}
+
+func (rh *remoteHypervisor) GetThreadIDs(ctx context.Context) (VcpuThreadIDs, error) {
+	// TODO
+	return VcpuThreadIDs{vcpus: make(map[int]int)}, nil
+}
+
+func (rh *remoteHypervisor) Cleanup(ctx context.Context) error {
+	// TODO
+	return nil
+}
+
+func (rh *remoteHypervisor) setConfig(config *HypervisorConfig) error {
+	// Create a Validator specific for remote hypervisor
+	rh.config = *config
+
+	return nil
+}
+
+func (rh *remoteHypervisor) GetPids() []int {
+	// TODO
+	return []int{99999}
```

Comment:
This pid is used by crio to fetch startTime, could you do please do something like:
```
diff --git a/src/runtime/virtcontainers/remote.go b/src/runtime/virtcontainers/remote.go
index 818da5167..ea1b3718f 100644
--- a/src/runtime/virtcontainers/remote.go
+++ b/src/runtime/virtcontainers/remote.go
@@ -8,6 +8,7 @@ import (
        "fmt"
        "net"
        "time"
+       "os"
 
        cri "github.com/containerd/containerd/pkg/cri/annotations"
        "github.com/containerd/ttrpc"
@@ -249,8 +250,8 @@ func (rh *remoteHypervisor) setConfig(config *HypervisorConfig) error {
 }
 
 func (rh *remoteHypervisor) GetPids() []int {
-       // TODO
-       return []int{99999}
+       // meanwhile let's use shim pid as it used by crio to fetch start time
+       return []int{os.Getpid()}
 }
 
 func (rh *remoteHypervisor) GetVirtioFsPid() *int {
 ```

Reply:
I've included this change.